functions/java-jre.bash

```diff
@@ -231,3 +238,78 @@ java_zulu_install_crypto_extension(){
   rm -rf "${policyTempLocation}"
   return 0
 }
+
+## Fetch AdoptOpenJDK using APT repository.
+##
+adoptopenjdk_fetch_apt(){
+  local adoptKey="/tmp/adoptopenjdk.asc"
+  echo -n "$(timestamp) [openHABian] Adding AdoptOpenJDK keys to apt... "
+  cond_redirect wget -qO "$adoptKey" https://adoptopenjdk.jfrog.io/adoptopenjdk/api/gpg/key/public
+  if cond_redirect apt-key add "$adoptKey"; then
+    echo "OK"
+  else
+    echo "FAILED (keyserver)"
+    rm -f "$adoptKey"
+    exit 1;
+  fi
+  echo -n "$(timestamp) [openHABian] Fetching AdoptOpenJDK... "
+  echo "deb http://adoptopenjdk.jfrog.io/adoptopenjdk/deb buster main" > /etc/apt/sources.list.d/adoptopenjdk.list
+  cond_redirect apt-get update
+  if cond_redirect apt-get install --download-only adoptopenjdk-8-hotspot-jre; then echo "OK"; else echo "FAILED"; exit 1; fi
+}
+
+## Install AdoptOpenJDK using APT repository.
+##
+adoptopenjdk_install_apt(){
+  if ! dpkg -s 'adoptopenjdk-8-hotspot-jre' > /dev/null 2>&1; then # Check if already is installed
+    echo -n "$(timestamp) [openHABian] Installing AdoptOpenJDK... "
+    adoptopenjdk_fetch_apt
+    cond_redirect systemctl stop openhab2.service
+    cond_redirect java_alternatives_reset
+    if cond_redirect apt-get install --yes adoptopenjdk-8-hotspot-jre; then echo "OK"; else echo "FAILED"; exit 1; fi
+    cond_redirect systemctl start openhab2.service
+  elif dpkg -s 'adoptopenjdk-8-hotspot-jre' >/dev/null 2>&1; then
+    cond_redirect systemctl stop openhab2.service
+    cond_redirect java_alternatives_reset
+    if cond_redirect dpkg-reconfigure adoptopenjdk-8-hotspot-jre; then echo "OK"; else echo "FAILED"; exit 1; fi
+    cond_redirect systemctl start openhab2.service
+  fi
+}
+
+## Reset Java in update-alternatives
+## Valid arguments: "Zulu"
```

Comment:
don't we need this for Adopt, too?
what if someone has Zulu installed and removes it?

Reply:
It checks to see if the folder is still there so that is no concern. I'm letting dpkg handle the AdoptOpenJDK install and registration for that.